flake.nix

```
@@ -0,0 +1,109 @@
+{
+  description = "Searx plugin for Searching the Green Web";
+
+  inputs.nixpkgs = {
+    type = "github";
+    owner = "NixOS";
+    repo = "nixpkgs";
+    ref = "21.05";
+  };
+
+  outputs = { self, nixpkgs }:
+    let
+      # System types to support.
+      supportedSystems = [ "x86_64-linux" ];
+      # Helper function to generate an attrset '{ x86_64-linux = f "x86_64-linux"; ... }'.
+      forAllSystems = f:
+        nixpkgs.lib.genAttrs supportedSystems (system: f system);
+      # Nixpkgs instantiated for supported system types.
+      nixpkgsFor = forAllSystems (system:
+        import nixpkgs {
+          inherit system;
+          overlays = [ self.overlay ];
+        });
+      in {
+
+        overlay = final: prev: rec {
+          tgwf-green-results-searx-plugin = prev.python38Packages.buildPythonPackage {
+            pname = "tgwf-searx-plugin";
+            version = "0.2";
+            buildInputs = [
+              prev.searx
+            ];
+            src = ./.;
+            doCheck = false;
+          };
+
+          # overlaying searx with version that depends plugin
+          # results in plugin being installed in the searx environment via python setuptools
+          # which is the requirement for the plugin to be available for searx:
+          # https://searx.github.io/searx/dev/plugins.html#external-plugins
+          searx = prev.searx.overrideAttrs (oldAttrs: {
+              propagatedBuildInputs = oldAttrs.propagatedBuildInputs ++ [
+                tgwf-green-results-searx-plugin
+              ];
+              postFixup = ''
+                ${oldAttrs.postFixup}
+                mkdir -p $out/share/static/plugins/external_plugins/plugin_only_show_green_results
+              '';
+            });
+
+        };
+
+        packages = forAllSystems (system:
+          {
+            inherit (nixpkgsFor.${system}) tgwf-green-results-searx-plugin searx;
+          }
+        );
```

Consider reducing amount of whitespace like so:
```suggestion
        packages = forAllSystems (system: {
            inherit (nixpkgsFor.${system}) tgwf-green-results-searx-plugin searx;
        });
```

---

I would agree with that formatting, `nixpkgs-fmt` disagrees
I propose to use auto-formatters output, that would result in more stable diffs